manifests/repo.pp

```diff
@@ -14,21 +14,28 @@
       Yumrepo['powerdns'] -> Package <| title == $powerdns::authoritative_package_name |>
       Yumrepo['powerdns-recursor'] -> Package <| title == $powerdns::recursor_package_name |>
 
+      $repo_desc = $facts['os']['release']['major'] ? {
+        '8'     => 'PowerTools',
+        default => 'CRB',
+      }
+      $repo_name = $repo_desc.downcase()
       if ($facts['os']['name'] == 'Rocky') {
-        $mirrorlist = "https://mirrors.rockylinux.org/mirrorlist?arch=\$basearch&repo=PowerTools-\$releasever"
-        $gpgkey = 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-rockyofficial'
+        $code_repo_params = {
+          mirrorlist => "https://mirrors.rockylinux.org/mirrorlist?arch=\$basearch&repo=${repo_desc}-\$releasever",
+          gpgkey     => 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-rockyofficial',
```

Comment:
I'd suggest to switch some generic name or leave the `centos` and `rocky` part out.

Reply:
I'm sure the other derivatives need these repos as well, however, I don't know enough about their structure to address it directly. The hiera based code makes no assumptions

Reply:
@saz handled this in a different PR